fix SDL_PumpEvents crash (still not implemented)

Cpasjuste opened the pull request.

xerpi commented:
I'm don't think it's a good idea to modify core SDL files, we should limit ourselves to vita specific files.

Cpasjuste commented:
I dont think SDL_GetKeyboardFocus is platform dependant so it could be tricky.

sezero committed:
bug #3739: handle %lu, %li and %ld in SDL_SetError.

slouken committed:
Add SDL_GetDisplayDPI implementation on Android. (thanks Rachel!)

slouken committed:
Added controller mapping for Android TV remotes
Also fixed the back button on the remote exiting the application

slouken committed:
Fixed building on non-Android platforms

slouken committed:
Fixed bug 3932 - Android, GetDisplayDPI release local reference

Sylvain

When writing JNI code, one has to make sure all local references are released otherwise the app end up crashing.

slouken committed:
Fixed bug 3926 - SDL_main export

e_pluschauskas

I noticed that after updating SDL to 2.0.6 my application now exports SDL_main.

At GitHub SDL mirror (branch 2.0.5) SDL_main prototyped as
extern C_LINKAGE int SDL_main(int argc, char *argv[]);
but at branch 2.0.6 prototype is
extern C_LINKAGE DECLSPEC int SDL_main(int argc, char *argv[]);

slouken committed:
Moved the meta-data into the right section

slouken committed:
Allow the activity to persist through connecting or disconnecting a keyboard

slouken committed:
Fixed bug 3917 - Android, issues with getManifestEnvironmentVariable

Sylvain

Some issue with this commit:
https://hg.libsdl.org/SDL/rev/4130b92b6be4

There is a memory allocation missing.

slouken committed:
Fixed bug 3933 - Android, no need of the listener thread

Sylvain

A listener thread has been added to know when the native thread would end.
But now, it is more easy to only check that after the main function has returned. It's one thread less.

slouken committed:
Fixed bug 3935 - Not find joysticks if android run 24-48 days.

Alexander Orefkov

In src\joystick\android\SDL_sysjoystick.c in SDL_SYS_JoystickDetect when SDL_GetTicks return number grater 2147483648 (after 24.85 days uptime) SDL_TICKS_PASSED(SDL_GetTicks(), timeout) return FALSE and Android_JNI_PollInputDevices is never calling.
And in JoystickByDeviceId - when search for newly added joystic - after SDL_SYS_JoystickDetect item not reinitilized, and always stay NULL, cause return NULL instead of added joystick.

slouken committed:
Make SDL fullscreen windows on Android actually fullscreen

olydis committed:
Emscripten: use cursor hotspot
the cursor's hotspot simply wasn't translated to it's CSS equivalent, yet
see https://developer.mozilla.org/en-US/docs/Web/CSS/cursor?v=example#Syntax. no explicit hotspot if (0|0) for compatibility with Edge and IE, which indeed don't support custom hot spots

slouken committed:
Android configure-based build fixes (thanks Vit!)

psyke83 committed:
SDL_video: try to bootstrap KMSDRM before RPI video driver
Allow better coexistence between RPI's vendor libraries and VC4 mesa driver.

psyke83 committed:
Raspberry Pi: fix ES 1/PVR support & autodetect Mesa driver
* The brcmGLESv2 vendor library also supports ES PVR/1 profiles
* Fallback to standard Mesa libraries if the VC4 driver is loaded

slouken committed:
Fixed bug 3940 - Add support for EGL_SURFACE_TYPE

tomwardio

Add support to be able to set EGL_SURFACE_TYPE bits when creating an EGL config. This is usefule when wanting to create pixel buffer surfaces in custom video drivers.

slouken committed:
Fixed bug 3917 - Android, issues with getManifestEnvironmentVariable

We're going to push the manifest environment variables from the Java side instead of continually querying for them from the native side.

slouken committed:
Fixed using libunwind on iOS, where it's only available on the simulator (thanks Vit!)

slouken committed:
Added float versions of SDL's math functions

slouken committed:
Updated math code from the uClibc 0.9.33 release

slouken committed:
Added SDL_fmod() and SDL_fmodf()

slouken committed:
Fixed compiler warnings after uClibc code update

slouken committed:
Fixed compiling on Windows with HAVE_LIBC enabled

slouken committed:
Updated version to 2.0.8 since SDL_image depends on it

slouken committed:
Fixed bug 3917 - Android, issues with getManifestEnvironmentVariable

Sylvain

What about getting some return code instead of creating another native function.

slouken committed:
Fixed bug 3939 - Remove static vm_error and vm_event from SDL_x11modes.c

tomwardio

Remove static int vm_error and vm_event, use local variables instead.

This fixes unused variable errors when compiling with SDL_VIDEO_DRIVER_X11_XINERAMA undefined.

src/video/x11/SDL_x11modes.c:505:22: error: unused variable 'vm_error' [-Werror,-Wunused-variable]

src/video/x11/SDL_x11modes.c:505:12: error: unused variable 'vm_event' [-Werror,-Wunused-variable]

slouken committed:
Updated version to 2.0.8 since SDL_image depends on it

icculus committed:
mouse: remove assert for unimplemented platforms (thanks, tomwardio!).

Fixes Bugzilla #3946.

BrandonSchaefer committed:
Fixed bug 3943 - General SDL_HINT_VIDEO_DOUBLE_BUFFER hint support

slouken committed:
Back out the following commits which are causing Dota 2 Exclusive Fullscreen to fail on AMD-based Macs.  The details of why this is being reverted are at: https://bugzilla.libsdl.org/show_bug.cgi?id=3949

Alex Szpakowski <slime73@gmail.com> 2017-07-12 21:28 -0300
macOS: Expose more display modes on retina screens. Fixes an issue found in BZFlag.
http://hg.libsdl.org/SDL/rev/1e26564c7288

Alex Szpakowski <slime73@gmail.com> 2017-07-12 21:32 -0300
Fix a potential crash in macOS 10.7 and earlier.
http://hg.libsdl.org/SDL/rev/424607440631

slime73 committed:
macOS: Fix an OS-generated warning printed to stdout on launch in bundled apps.

slouken committed:
Replaced the intrinsics header with the single header that includes all intrinsics so we can use AVX, etc.

slouken committed:
Only include Intel intrinsics header on x86 and x64

slouken committed:
Fixed typo in comment

slouken committed:
Fixed Android build error on older SDK

slouken committed:
Updated SDL's YUV support, many thanks to Adrien Descamps

New functions get and set the YUV colorspace conversion mode:
	SDL_SetYUVConversionMode()
	SDL_GetYUVConversionMode()
	SDL_GetYUVConversionModeForResolution()

SDL_ConvertPixels() converts between all supported RGB and YUV formats, with SSE acceleration for converting from planar YUV formats (YV12, NV12, etc) to common RGB/RGBA formats.

Added a new test program, testyuv, to verify correctness and speed of YUV conversion functionality.

rsn8887 committed:
merge latest changes from upstream (spurious)

rsn8887 committed:
vita: fix merge

slouken committed:
Fixed building YUV test programs (thanks Ozkan!)

slouken committed:
Fixed building yuv_rgb.h when stdint.h isn't available (thanks Ozkan!)

slouken committed:
Fixed bug 3964 - YUV to RGB in video/SDL_yuv.c is broken for any output format of type ABGR8888 or BGR888

raist66676

Here is the bug in latest SDL 2.0.8 development repo. It is obvious and simple to fix by correcting typos on six lines of code.

In src/video/SDL_yuv.c on lines 217, 249, 280, 321, 353, and 384 the wrong conversion functions are called for SDL_PIXELFORMAT_ABGR8888 and SDL_PIXELFORMAT_BGR888. Instead of ABGR functions, BGRA functions are called. These are typos.

rsn8887 committed:
Merge branch 'master' of github-rsn8887:spurious/SDL-mirror

rsn8887 committed:
Vita: fix merge

slouken committed:
Added support for aarch64 memory barrier instruction

slouken committed:
Fixed bug 3959 - cmake build broken by commit 11702 (cf166abbde4a) due to typo

Mate Nagy
There is a typo in CMakeLists.txt that makes CMake exit with failure.

Change that causes the problem: (Notice the double ending brackets)
   ${SDL2_SOURCE_DIR}/src/video/*.c)
+  ${SDL2_SOURCE_DIR}/src/video/yuv2rgb/*.c)

Fix:
Just remove the first ending bracket resulting in:
   ${SDL2_SOURCE_DIR}/src/video/*.c
   ${SDL2_SOURCE_DIR}/src/video/yuv2rgb/*.c)

slouken committed:
Fixed bug 3973 - Include of stdint is needed to build on linux, mingw and possibly android

Stuart Axon

https://discourse.libsdl.org/t/stdint-h-removed/23426

https://discourse.libsdl.org/t/debuild-fails-to-build-the-last-few-days/23429/4

Currently SDL2 is not building in Linux (x86 and ARM), Android and MingW because include stdint.h has been removed from yuv_rgb.h

Cpasjuste committed:
Merge pull request #10 from rsn8887/master

merge

flibitijibibo committed:
Steam Virtual GamePad config for macOS

slouken committed:
Fixed bug 3976 - SDL drivers may leak driverdata memory due to ignoring return value of SDL_AddDisplayMode

C Snover

SDL_AddDisplayMode returns an SDL_bool corresponding to whether or not the given display mode was added or not. It will return SDL_FALSE if a matching display mode already exists in the display's list of display modes, which causes ownership of the mode driverdata to remain with the caller. Some video drivers ignore the return value of SDL_AddDisplayMode, so leak the driverdata memory when SDL_AddDisplayMode returns SDL_FALSE.

slouken committed:
Disabled spurious Visual Studio warnings in the uClibc math code

slouken committed:
Fixed some compiler warnings

slouken committed:
Fixed bug 3979 - Failed to link with "multiple definition of `scalbln'" error on both MinGW and MinGW-w64 by CMake

Vitaly Novichkov

Once I ran build of my codecs collection on AppVeyor where my CMake script downloads latest SDL2 from HG repo, failed to link because of math functions conflict:

https://ci.appveyor.com/project/Wohlstand/audiocodecs/build/1.0.44
The revision is b94292eebb5c

There are both vanilla MinGW and MinGW-w64 are failed to build.


```
[100%] Linking C shared library libSDL2.dll
c:/mingw/bin/../lib/gcc/mingw32/5.3.0/../../../libmingwex.a(scalbn.o):(.text+0x0): multiple definition of `scalbln'
CMakeFiles\SDL2.dir/objects.a(s_scalbn.c.obj):C:/projects/audiocodecs/build-MinGW-Release-Win32/external/SDL2/src/SDL2HG/src/libm/s_scalbn.c:30: first defined here

slouken committed:
Fixed bug 3980 - Fix for KMSDRM driver where cursor would not be shown on some gfx hardware because of unsupported cursor size

Manuel Alfayate Corchete

This fixes a problem with KMSDRM on some graphics hardware where only bigger cursor sizes are supported, such as current Intel gfx. (The kernel-side driver is what limits this: had to look for failing IOCTLs...)
That caused SDL_SetCursor() to fail silently, and we were left with a missing cursor without further explanation.
With this patch, different "standard" sizes are tried and a bigger one is used (with an intermediate and clean buffer only used to write the new cursor to the BO where it will live after) if we get, let's say, 16x16 which is pretty common but our hardware does not support that.

slouken committed:
Use the included Khronos headers on Android so we can create Core OpenGL contexts when building with older SDK

slouken committed:
Non-resizable windows need to have their window rect set to the client rect

slouken committed:
Fixed bug 3945 - Add eglCreatePbufferSurface function

tomwardio

Proposed patch loads eglCreatePbufferSurface in same manner as other 1.1 functors. This allows custom video drivers to create pbuffer surfaces.

slouken committed:
Fixed name of eglCreatePbufferSurface function

suikki committed:
Fixed SDL_Log not working on windows if the output is being redirected.

slouken committed:
Fixed typos (thanks Martin!)

slouken committed:
Removed spurious debug output

slouken committed:
Fixed bug 3975 - Add GLES2 support for macOS via ANGLE library

Andrey

Seems latest google angle library successfully built & tested under macOS'es.

https://github.com/google/angle

We need to use GLES2 to implement true cross-platform code.

slouken committed:
Fixed bug 3974 - Fix SDL_WarpMouseInWindow on both KMSDRM and RaspberryPi drivers

Manuel Alfayate Corchete

This patch fixes SDL_WarpMouseInWindow() in both the KMSDRM and Raspberry Pi graphic backends.

Cpasjuste committed:
Merge pull request #11 from spurious/master

spurious merge

icculus committed:
configure: List Metal in video drivers.

--HG--
extra : rebase_source : f5a3e33689d899707204aafed145cdf4ba0c389e

icculus committed:
metal: Don't check if Metal is available if targeting modern macOS versions.

--HG--
extra : rebase_source : f007831a3ff9aebb17ec3529e6fee4c3ad63d0b1

slouken committed:
Fixed bug 1878 - Scaled texture draws with filtering produce wrapping artifacts.

Yuri K. Schlesner

When using texture filtering, there are filtering artifacts visible on the edges of scaled textures, where the texture filtering pulls in texels from the other side of the texture. Using clamping texture modes wouldn't completely fix this since source rectangles don't need to cover the whole texture. (See screenshot attached in next post.)

The opengl driver uses clamping on textures and so avoid this at least in the cases where the source rect is the whole texture. The direct3d driver does not and so has problems in every case. I'm not sure if it can actually completely be fixed, but at least enabling clamping for direct3d would be one step in the right direction.

slouken committed:
Fixed compiler warning

slouken committed:
Added check for failure of D3D_ActivateRenderer()

slouken committed:
Fixed minor memory leak in the Metal renderer

slouken committed:
Minor cleanup

slouken committed:
Fixed Metal renderer memory leak

slouken committed:
Added SDL_RenderGetMetalLayer() and SDL_RenderGetMetalCommandEncoder()

icculus committed:
hints: Add "metal" to the list of SDL_RENDER_DRIVER hints.

--HG--
extra : rebase_source : f474551b0c211cb7dbd5ee610692c728b24b701a

icculus committed:
metal: Cleaned up some reference count politics.

--HG--
extra : rebase_source : 96d3ab1460cf610112d7fcbfc056c6a9faec3455

slouken committed:
Removed duplicate entry for SDL_wasapi.h

Cpasjuste committed:
PS3: first commit

icculus committed:
audio: fixed typo in Doxygen comment.

--HG--
extra : rebase_source : daf1f903439d66b9e81d35831fb72960ced55111

icculus committed:
metal: Added some comments and FIXMEs.

--HG--
extra : rebase_source : 6a5b3bcfbb43cc34b72d646617c09af26f16632e

icculus committed:
metal: fixed render target support.

--HG--
extra : rebase_source : 0d438b9eca6be9264b9f005beba23bdee94eb884

slouken committed:
Fixed pixel positioning and size for the Metal renderer

slouken committed:
Fixed compiler warning

slouken committed:
Added support for linear sampling and pixel coordinates in the metal renderer

slouken committed:
Fixed normalized coordinates when the viewport is set

slouken committed:
Backed out using pixel texture coordinates, it had weird visual side effects

slouken committed:
Added the Metal framework to several iOS tests

slouken committed:
Added SDL_WinRTGetDeviceFamily() to find out what type of device your application is running on (thanks Daniel Knobe!)

slouken committed:
Workaround for bug 3931 - spurious SDL_MOUSEMOTION events with SDL_HINT_MOUSE_RELATIVE_MODE_WARP 1 since Windows 10 Fall Creators update

Elisée Maurer

The attached minimal program sets the SDL_HINT_MOUSE_RELATIVE_MODE_WARP to 1, enables relative mouse mode then logs all SDL_MOUSEMOTION xrel values as they happen.

When moving the mouse exclusively to the right:

 * On a Windows 10 installation before Fall Creators update (for instance, Version	10.0.15063 Build 15063), only positive values are reported, as expected
 * On a Windows 10 installation after Fall Creators update (for instance, Version 10.0.16299 Update 16299), a mix of positive and negative values are reported.

3 different people have reproduced this bug and have confirmed it started to happen after the Fall Creators update was installed. It happens with SDL 2.0.7 as well as latest default branch as of today.

It seems like some obscure (maybe unintended) Windows behavior change? Haven't been able to pin it down more yet.

(To force-upgrade a Windows installation to the Fall Creators update, you can use the update assistant at https://www.microsoft.com/en-us/software-download/windows10)

Eric Wasylishen

Broken GetCursorPos / SetCursorPos based games on Win 10 fall creators are not limited to SDL.. I just tested winquake.exe (original 1997 exe) and it now has "jumps" in the mouse input if you try to look around in a circle. It uses GetCursorPos/SetCursorPos by default. Switching WinQuake to use directinput (-dinput flag) seems to get rid of the jumps.

Daniel Gibson

A friend tested on Win10 1607 (which is before the Fall Creators Update) and the  the bug doesn't occur there, so the regression that SetCursorPos() doesn't reliably generate mouse events was indeed introduced with that update.
I even reproduced it in a minimal WinAPI-only application (https://gist.github.com/DanielGibson/b5b033c67b9137f0280af9fc53352c68), the weird thing is that if you don't do anything each "frame" (i.e. the mainloop only polls the events and does nothing else), there are a lot of mouse events with the coordinates you passed to SetCursorPos(), but when sleeping for 10ms in each iteration of the mainloop, those events basically don't happen anymore. Which is bad, because in games the each iteration of the mainloop usually takes 16ms..

I have a patch now that I find acceptable.
It checks for the windows version with RtlGetVersion() (https://msdn.microsoft.com/en-us/library/windows/hardware/ff561910.aspx) and only if it's >= Win10 build 16299, enables the workaround.
All code is in video/windows/SDL_windowsevents.c
and the workaround is, that for each WM_MOUSEMOVE event, "if(isWin10FCUorNewer && mouseID != SDL_TOUCH_MOUSEID && mouse->relative_mode_warp)", an addition mouse move event is generated with the coordinates of the center of the screen
(SDL_SendMouseMotion(data->window, mouseID, 0, center_x, center_y);) - which is exactly what would happen if windows generated those reliably itself.
This will cause SDL_PrivateSendMouseMotion() to set mouse->last_x = center_x; and mouse->last_y = center_y; so the next mouse relative mouse event will be calculated correctly.

If Microsoft ever fixes this bug, the IsWin10FCUorNewer() function would have to
be adjusted to also check for a maximum version, so the workaround is then disabled again.

slouken committed:
Fixed Mac OS X build

slouken committed:
The newer compilers generate the property boilerplate automatically

slouken committed:
Fixed Windows gcc build

slouken committed:
Check for immintrin.h before using it in SDL_cpuinfo.h

slouken committed:
Fixed bug 3999 - Build failed when try to build with configure script for Android

slouken committed:
Added SDL_uikitmetalview.m to the tvOS project

slouken committed:
Renamed Xcode build targets so it's easier to copy files out of build directories on the command line
Also enabled bitcode, which is needed for App Store builds

slouken committed:
Merged latest changes from rel/streaming_client

slouken committed:
Added support for Android video textures

slouken committed:
Add the ability to set SDL to handle Return as 'hide IME' on Android softkeyboard. (thanks Rachel!)

slouken committed:
Fixed bug 3996 - Corrupted or over-released critical section on SDL_Quit (SDL_event_watchers_lock)

Andrew

This likely comes down to an additional 'unlock' being called before destroying it, without a matching 'lock'.

slouken committed:
Fixed bug 3993 - altivec.h include in SDL_cpuinfo.h breaks compilation with -std=c++11

bastien.bouclet

According to this GCC bug report, altivec.h requires building with the gnu extensions: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=78263.

As an application developer, I don't want SDL to force me to enable the gnu extensions.

slouken committed:
Fixed bug 3981 - Inverted logic bug in SDL_renderer "overscan" feature

Eric wing

There is a tiny bug in the new overscan code for the SDL_renderer.

In SDL_renderer.c, line 1265, the if check for SDL_strcasecmp with "direct3d" needs to be inverted.

Instead of:
if(SDL_strcasecmp("direct3d", SDL_GetCurrentVideoDriver())) {

It should be:
if(0 == SDL_strcasecmp("direct3d", SDL_GetCurrentVideoDriver())) {

This bug causes the "overscan" mode to pretty much be completely ignored in all cases and all things remain letterboxed (as before the feature).

slouken committed:
Fixed bug 3992 - SDL_GetColorKey doesn't set error message

Luke A. Guest

SDL_GetColorKey does not set an error message on failure. The current source just returns -1.

The documentation https://wiki.libsdl.org/SDL_GetColorKey?highlight=%28%5CbCategoryAPI%5Cb%29%7C%28SDLFunctionTemplate%29 says to call SDL_GetError but that is useless in this case.

icculus committed:
windows: Fixed a comment.

--HG--
extra : rebase_source : 94621aac2da4199e3f79fb63c3bfd8db49d63a4e

slouken committed:
Parameterized the All (target platform) build target scripts so they can be used in other projects

slouken committed:
Moved deployment target values to project settings

icculus committed:
wasapi: switched to event-driven interface.

This reduces latency and improves battery life.

--HG--
extra : rebase_source : 94378e202c98258e9b59c454a8c81fe11470b956

slouken committed:
Added Android build script for SDL

slouken committed:
Added support for the ASUS TV500BG Android gamepad

slouken committed:
Fixed handling of the Apple TV remote buttons
The menu button on the remote is the back button

slouken committed:
Fixed default Android D-PAD mapping so the back button is treated as B

slouken committed:
Fixed bug 4005 - Android, SDL_IsGameController() crashes is index is out of range

Sylvain

On Android, if you give an invalid index to SDL_IsGameController(), it will crash in SDL_SYS_IsDPAD_DeviceIndex().

slouken committed:
Fixed bug 4004 - iOS: don't hide keyboard on RETURN

Dominik Reichardt

As discussed in 2012 the iOS onscreen keyboard hides when you hit RETURN (see https://discourse.libsdl.org/t/on-screen-keyboard-change/19216).
IMO this is a bad idea to not be able to influence this behavior and just recently this was fixed for Android by adding the hint SDL_HINT_ANDROID_RETURN_HIDES_IME in changeset 11768	0f2fb9ec6ad6.

slouken committed:
Fixed bug 4003 - HAVE_POLL undefined in SDL_poll.c, making it impossible to use.

tomwardio

HAVE_POLL is correctly defined in SDL_config.h when running configure. However, in the only place where it's used, it's undefined at the start of the file.

slouken committed:
Fixed bug 4000 - SDL2 on raspberry: mouse displayed at 0,0 after SDL_ShowCursor

Laurent Merckx

I have a problem with the SDL_ShowCursor method on Raspberry.

Depending on the context, my application hides or show the mouse cursor with SDL_ShowCursor.
But when calling SDL_ShowCursor(true), the cursor is displayed at 0,0 (and not at last position).

After debugging sources by myself, it seems that the problem is in SDL_rpimouse.c - RPI_ShowCursor:

vc_dispmanx_rect_set( &dst_rect, 0, 0, curdata->w, curdata->h);
should be
vc_dispmanx_rect_set( &dst_rect, mouse->x, mouse->y, curdata->w, curdata->h);

For me, it solves the problem.

slouken committed:
Fixed bug 4001 - Android, prevent error message box to crash

Sylvain

Prevent the error message box to crash after being clicked.
Because of "SDLActivity.mSingleton.finish();"

slime73 committed:
macOS: Make sure the desktop's display mode is always in SDL's list of display modes.

icculus committed:
xcode: Add Metal and QuartCore to everything as a weak_framework.

--HG--
extra : amend_source : 351d631ed41aae1f7a0ff8f8875958abf9b99808

slime73 committed:
metal: only 4 (instead of 5) vertices are needed to draw rectangles and textures, and only 3 (instead of 5) vertices are needed to cover the screen for the 'full-screen quad' when clearing.

slime73 committed:
metal: use a projection matrix instead of manually transforming vertices into clip space on the CPU.

slime73 committed:
metal: Respect the vsync flag on macOS 10.13+.

icculus committed:
winrt: bumped SDL's shipping version number in winrtbuild.ps1

icculus committed:
winrt: Disabled Windows Phone 8.0 builds.

It lacks APIs we need for WASAPI and probably no one will miss it. 8.1 is
still supported.

icculus committed:
wasapi: Patched to compile on non-UWP WinRT builds.

icculus committed:
windows: Use WaitForSingleObjectEx() always

This is available since Windows XP, so it's safe to use always, not just in
a WinRT ifdef.

icculus committed:
winrt: workaround for pre-UWP builds.

There's probably a better way to do this for legacy platforms, though.

icculus committed:
winrt: Add mmdevapi.lib dependency to WinRT8.1 builds, for WASAPI support.

slime73 committed:
macOS: Fix MoltenVK Metal view resizing, and allow the metal view to be used without vulkan.

slime73 committed:
metal: Use the existing cocoa code for creating a Metal view on macOS. Fixes the renderer size when the window is resized.

slime73 committed:
metal: implement SDL_RenderCopyEx, and fix a memory leak in SDL_CreateTexture.

slime73 committed:
metal: Add support for custom blend modes.

slime73 committed:
metal: Use sampler state objects instead of shader-declared samplers for linear vs nearest filtering.

This avoids a ton of shader duplication once multiple shaders that use samplers are added (e.g. the currently missing YUV shaders).

slime73 committed:
iOS: fix build

icculus committed:
windows: Remove references to GetVersionExA (thanks, Andrew Pilley!).

"GetVersionExA is deprecated in windows 8.1 and above's SDK, causing a warning
when building against the win10 SDK. Attached patch cleans up the usage for a
warning-free build.

GetVersionExA was being used to test to see if SDL was running on win9x or
winnt. A quick chat with Ryan on twitter suggested that SDL doesn't
officially support win9x anymore, so the call to this can be outright removed.

As an aside, replacing the call to GetVersionExA with VerifyVersionInfoA (the
recommended path) would have been pointless, as VerifyVersionInfoA only
supports VER_PLATFORM_WIN32_NT and doesn't officially support any other value
for dwPlatformId currently. (And it's probable that win9x SDKs didn't have
VerifyVersionInfo* in them anyway.)"

Fixes Bugzilla #4019.

slime73 committed:
metal: Clean up manual reference counting. Fixes some memory leaks.

slime73 committed:
metal: Fix a typo preventing iOS compilation...

slouken committed:
Fixed metal renderer pixel centers when drawing

slouken committed:
Fixed direction of y adjustment for new orthographic projection in the metal renderer

slime73 committed:
configure script: Implement testing for build-time Metal SDK support.

slime73 committed:
metal and moltenvk: fix highdpi.

slime73 committed:
metal: Misc. improvements.

- Use a single buffer for various non-changing constants accessed by the GPU, instead of multiple buffers.
- Do the half-pixel offset for points and lines using a transform matrix so we don't need a malloc when rendering.
- Don't add a half-pixel offset for other primitives and textures. This matches D3D and GL render behaviour.
- Remove the half-texel texture coordinate offset since it's not needed now that there's no more half-pixel position offset when rendering a texture.
- Don't try to set texture usage on iOS 8 since it doesn't exist there.

slouken committed:
Updated copyright for 2018

slouken committed:
Fixed bug 4011 - Wayland: fix free cursor

Vladimir

Sometimes SDL application crashes on cursor free.

Patch tested under Centos 7 + weston

slouken committed:
Fixed bug 4012 - Wayland: invalid direction on mouse wheel

Vladimir

Invalid direction on mouse wheel

Patch tested in Centos 7 + Weston

slouken committed:
Fixed bug 4013 - Wayland: fix videoquit on multimonitor system

Vladimir

On multimonitor system Wayland_VideoQuit invalid deiniting.

Tested in Centos7 + Weston

slouken committed:
Fixed bug 4018 - Implement SDL_GetWindowBordersSize() under Windows/Win32/WinAPI

Ismael Ferreras Morezuelas (Swyter)

As a new year gift I have implemented the Windows version of SDL_GetWindowBordersSize(). I needed it for auto-selecting a cozy window size for the game I'm currently working on and noticed that it only worked under X11, so I thought it could be a good excuse to contribute back more stuff. The Mercurial patch is attached as a .diff file. Let me know what you think.

Happy 2018 to all the SDL2 devs and users!

--

PS: Keep in mind that Windows 10 includes the 8px invisible grip borders as part of the frame. There's a way of detecting if Aero/DWM is being used and ask only for the visible rect, but I believe that GetWindowRect() is doing that for a reason and working as intended, so I haven't changed it. (See [2])


References:
[1]: http://www.firststeps.ru/mfc/winapi/r.php?72
[2]: https://stackoverflow.com/a/34143777/674685
[3]: https://stackoverflow.com/a/431548/674685
[4]: https://wiki.libsdl.org/SDL_GetWindowBordersSize

slouken committed:
Fixed spacing in copyright headers

slime73 committed:
metal: Implement fast hardware clearing when possible, by deferring the start of a render pass until a clear or draw operation happens.

slime73 committed:
metal: use a private instead of managed buffer for the renderer's non-changing constant data.

Recommended by Xcode's Metal frame capture analysis.

slime73 committed:
metal: Add support for YUV/NV12 texture formats.

slime73 committed:
metal: Fix pipeline states to use the pixel format of the current render target, instead of a hard-coded format.

slime73 committed:
metal: set max texture size based on device capability.

tlgkccampbell committed:
Allow Android Java shim to be built as an AAR

slouken committed:
Added availability check to fix compiler warning for symbol only available on tvOS 11.0 and newer

slime73 committed:
Android: resolve symlinks in SDL_AndroidGetInternalStoragePath (thanks Henrique Gemignani and cigumo!)

Fixes issues on modern Android versions when the path is used in code that explicitly doesn't follow symlinks (such as PHYSFS_mkdir).

slouken committed:
Restored borderless window behavior where DOTA created a borderless window the size of the desktop and expected it to behave like a fullscreen desktop window.

A future SDL release will change the borderless window to act more like a normal window that happens to have no chrome, to support windows that draw their own chrome. In the meantime, those applications should set the "SDL_BORDERLESS_WINDOWED_STYLE" hint.

mariogrip committed:
Mir: Handle close window events

johnbartholomew committed:
Vulkan: Allow SDL_Vulkan_GetInstanceExtensions to be called with a larger array than necessary.

slouken committed:
Fixed bug 4043 - SDL_windowswindow.c incorrect icon height

Needed to allocate space for the mask in the ICONIMAGE structure

deveee committed:
EGL: Request sRGB framebuffer in correct place.

The EGL_GL_COLORSPACE_KHR is an attribute for eglCreate*Surface.

As written in EGL_KHR_gl_colorspace documentation:

    Accepted as an attribute name by eglCreateWindowSurface,
    eglCreatePbufferSurface and eglCreatePixmapSurface

        EGL_GL_COLORSPACE_KHR                   0x309D
    (...)

flibitijibibo committed:
SDL_log10

slouken committed:
Fixed formatting, added actual count to SDL error message

gamaral committed:
Make rpi video cross-compiler friendly.
* Stops using fixed path to find GLES/EGL libs.
* Tries pkg-config to locate bcm_host.

Signed-off-by: Guillermo A. Amaral <g@maral.me>

slouken committed:
Added a hint SDL_HINT_APPLE_TV_REMOTE_SWIPES_AS_ARROW_KEYS to prevent turning Apple TV remote swipes into arrow key events

icculus committed:
wasapi: Fixed some compiler warnings.

icculus committed:
android: Fixed compiler warning about nested '/*' comments.

icculus committed:
testoverlay2: use SDL_atoi, not atoi.

slouken committed:
Fixed bug 3985 - SDL_CreateWindow() has stopped changing screen mode when SDL_WINDOW_FULLSCREEN is used

Anthony

This worked in 2.0.5 as normal, but stopped working in 2.0.7. The monitor's resolution doesn't change, a window is created in full screen mode at the virtual desktop resolution instead.

slouken committed:
Fixed bug 4065 - SDL_wave.c fail to compile in armeabi with ndk r16b

Sylvain

armeabi is almost deprecated for android-ndk higher that r13b.
either it doesn't compile (ICE), or it executes wrongly (using long long for instance).

android people advices to use armeabi-v7a (or use r13b).

slouken committed:
Removed armeabi from the SDL build API list, since it's not supported by the latest Android SDK

slouken committed:
Fixed misleading indentation

slouken committed:
Fixed ISO C99 compatibility
SDL now builds with gcc 7.2 with the following command line options:
-Wall -pedantic-errors -Wno-deprecated-declarations -Wno-overlength-strings --std=c99

slouken committed:
Fixed building on platforms without __sighandler_t

slouken committed:
Added SDL_HINT_IOS_HIDE_HOME_INDICATOR to determine how the home indicator on the iPhone X is handled.

This variable can be set to the following values:
   "0"       - The indicator bar is not hidden (default for windowed applications)
   "1"       - The indicator bar is hidden and is shown when the screen is touched (useful for movie playback applications)
   "2"       - The indicator bar is dim and the first swipe makes it visible and the second swipe performs the "home" action (default for fullscreen applications)

slouken committed:
Fixed building on tvOS

slouken committed:
SDL Changes to support clean reads
CR: saml

slouken committed:
Added SDL_IsAndroidTV()

slouken committed:
Replaced SDL_HINT_APPLE_TV_REMOTE_SWIPES_AS_ARROW_KEYS with SDL_HINT_TV_REMOTE_AS_JOYSTICK which controls whether remotes on iOS and Android are interpreted as joysticks (the default) or as return/escape/arrow keys.

slouken committed:
Send Apple TV remote input as key events unless it's opened as a joystick, to match Android behavior.

icculus committed:
cmake: Minor cleanup (reuse a string we already calculated).

--HG--
extra : rebase_source : 06264d613a3b7a95be897e4acd91af38afba21e9

icculus committed:
wayland: Add support for xdg-shell protocol (unstable v6).

This is meant to be the desktop-enhanced version of wl_shell. Right now we
just match what the existing wl_shell code does, but there are other areas of
functionality available to us now, that we can fill in later.

This uses the "unstable" API, since this is what ships in Ubuntu 17.10 (as
part of Wayland 1.10), but Wayland 1.12 promotes this to stable with extremely
minor changes. We will add support for the stable version when it makes sense
to do so.

--HG--
extra : rebase_source : dbc84f58501611364f8ecabe5a004e26b4debbf1

slouken committed:
Description: fix installation of shared library for Android
Author: Boris Pek <tehnick-8@yandex.ru>
Last-Update: 2018-01-10

slouken committed:
Fixed bug 4060 - Alternative DualShock 4 v2 controller GUID

ayer.3d

I have a DualShock 4 v2 controller with a GUID that's not in the database. There is an existing GUID that is almost identical, with the only difference that I can tell being the reported version string (mine being 8001, database is 8100).

Existing GUID: 050000004c050000cc09000000810000
     New GUID: 050000004c050000cc09000001800000

When connected via USB, the GUID matches an existing entry: 030000004c050000cc09000011810000

gamaral committed:
Set rpath instead of changing environment for RPi
Credit goes to Adrian Perez de Castro for the improvement.

Signed-off-by: Guillermo A. Amaral <g@maral.me>

slouken committed:
Fixed bug 4054 - Raspberry Pi refresh rate detection

Viacheslav Slavinsky

SDL_rpivideo driver has 60 frames per second hardcoded in it, this is a problem for games that need to keep pace using VSYNC. I believe that I have found a solution to this. It is based on code in tvservice.c in rpi userland:

https://github.com/raspberrypi/userland/blob/a1b89e91f393c7134b4cdc36431f863bb3333163/host_applications/linux/apps/tvservice/tvservice.c#L433

slouken committed:
Fixed bug 4021 - Android, hard-coded Keycode value

Sylvain

There is an hard-coded keycode value in SDLActivity.java

sezero committed:
add SDL_log10 and SDL_log10f to include and dynapi

slouken committed:
Fixed min/max window size handling for borderless resizable windows

slouken committed:
Windows Phone 8.0 is no longer supported

slouken committed:
Fixed compiler warning on Android

slouken committed:
Fixed compile warning

slouken committed:
Fixed bug 4073 - Unquoted Unicode argument parsing broken on Windows due to incorrect usage of SDL_isspace()

icculus committed:
vulkan: Fix assignment of vkGetInstanceProcAddr on Windows.

"*(void**)pfn = LoadAddress()" would cast the NULL pointer in pfn to a
void**, and then dereference it, which wasn't what we wanted. Replaced with
a clearer cast operation.

--HG--
extra : rebase_source : 6dd9a59b02023162ab0faa8c7c300d881fdcfcf9

slouken committed:
ISO C correct fix for casting void* to function pointer

slouken committed:
Fixed bug 4075 - configury adds Metal.framework to linkage even if it is not available

Ozkan Sezer

Configury adds Metal.framework to linkage even if it is not available.
My solution is setting enable_render_metal to no when Metal.framework
is not found

slouken committed:
On Android show the system UI when an SDL window is windowed, hide the system UI when it's fullscreen, like we do on iOS.
We're increasing the Android SDK minimum version to API 19, this doesn't increase the minimum target API, which is API 14.

sezero committed:
kill C99'ism in SDL_waylandvulkan.c

sezero committed:
fix building SDL_audiotypecvt.c with gcc < 4.0

sezero committed:
revert the recent typecast assignment changes (see bug #4079)

also change the void* typedefs for the two vulkan function
pointers added in vulkan_internal.h  into generic function
pointer typedefs.

slouken committed:
Fixed bug 4081 - sdlchecks.cmake: typo introduced by rev11848 breaks building with wayland support

slouken committed:
Fixed bug 3950 - Don't always call dbus_shutdown in SDL_DBus

Alexander Larsson

dbus_shutdown() is a debug feature which closes all global resources in the dbus library. Calling this should be done by the app, not a library, because if there are multiple users of dbus in the process then SDL could shut it down even though another part is using it.

For example, i had an issue with this in mGBA, which uses both Qt and SDL, both using libdbus. I had a session bus, but no system bus (this was in a flatpak sandbox), and when SDL_DBus_Init() failed to init the system bus it called dbus_shudown() and continued on. This caused issues for Qt when running due to its session bus connections having disappeared beneath it.

slouken committed:
Fixed bug 3947 - replace strlcpy with memcpy in SDL_strdup()

slouken committed:
Fixed bug 3920 - IBus not work with SDL 2.0.7

cjacker

After updating from 2.0.5 to 2.0.7, Ibus not work anymore(fcitx still works).

Compare with 2.0.5, there are two issues in SDL_ibus.c.

1, SetupConnection always return SDL_FALSE in 2.0.7.

2, 'SetCapabilities' method should be called on 'ibus_conn'.

Patch attached.

slouken committed:
Fixed bug 4027 - CheckLibSampleRate macro in sdlchecks.cmake never defines HAVE_LIBSAMPLERATE{,_SHARED}, so they're always reported as disabled by MESSAGE_TESTED_OPTION macro in macros.cmake

slouken committed:
Don't attempt WM_NCCALCSIZE adjustment when in fullscreen window transition

slouken committed:
Fixed bug 4085 - X11: Allow configuring _NET_WM_BYPASS_COMPOSITOR through SDL hints

Callum McGing

This patch allows the user to disable the behaviour that blocks the compositor through a new hint: SDL_VIDEO_X11_NET_WM_BYPASS_COMPOSITOR. This allows tools or other windowed applications to behave properly under KWin.

icculus committed:
video: put a spinlock around a global linked list.

This should only contend if you're allocating or freeing surfaces from
multiple threads at once, and then just for a short time.

Fixes Bugzilla #4084.

--HG--
extra : rebase_source : d83a60e93030214968dfad99d54fb9c79a1fa97a

icculus committed:
pulseaudio: Just read/dump captured data in FlushCapture.

Apparently pa_stream_flush() doesn't work as expected:

https://lists.freedesktop.org/archives/pulseaudio-discuss/2012-April/013328.html

Fixes Bugzilla #4087.

--HG--
extra : amend_source : fd1141d8f1df793c80561e0de05a5c7a15950568

icculus committed:
yuv: fixed variable declaration shadowing warnings.

Fixes Bugzilla #4062.

--HG--
extra : rebase_source : 3bb1d5bb198a2f9ec55ac2006569e80205e4a119

icculus committed:
yuv: patched to compile.

icculus committed:
pthread: fix error code checks (thanks, Andreas!).

Most pthread functions return 0 on success and non-zero on error, but those
errors might be positive or negative, so checking for return values in the
Unix style, where errors are less than zero, is a bug.

Fixes Bugzilla #4039.

slouken committed:
Fixed bug 4034 - Do we really need _DllMainCRTStartup() in every Windows build?

Andreas Falkenhahn

In src/SDL.c there is this code:

_DllMainCRTStartup(HANDLE hModule,

...

The comment says that this is needed on Watcom C for some reason but why is it included then when building with Visual C as well? Shouldn't it be only included when compiling on Watcom C then?

I'm asking because this code caused me a lot of headaches because I'm building a DLL that contains SDL and I link using /MT and the _DllMainCRTStartup() symbol obviously led to lots of trouble but it wasn't clear to me where the problem was because all I got from the linker was:

LNK2019: unresolved external symbol _main referenced in function ___tmainCRTStartup

So I had to got through each and every object to see what the culprit was. See here for the full story:
https://stackoverflow.com/questions/25067151/lnk2019-unresolved-external-symbol-main-referenced-in-function-tmaincrtstar/48177067#48177067

So if it isn't necessary on Visual C, please just leave that symbol out on Visual C so that it no longer leads to any trouble. Thanks.

slouken committed:
Actually, this is needed for building with Visual Studio with both /MT and /MD.
With the previous change, I get:
1>     Creating library C:\projects\SDL\VisualC\Win32\Debug\SDL2.lib and object C:\projects\SDL\VisualC\Win32\Debug\SDL2.exp
1>LINK : error LNK2001: unresolved external symbol __DllMainCRTStartup@12

slouken committed:
Fixed bug 4088 - Fix Metal link errors with test programs in SDLTest.xcodeproj

Eric Wasylishen

The following patch adds Metal.framework to the "link binary with libraries" section of each test program, with "status" set to "optional", which fixes link errors on all of the test programs. I'm not sure if this is a correct fix - the fact that this was necessary might indicate the static SDL2.a library has a hard dependency on Metal.framework (?) - but it gets the test programs working in Xcode again.

It also adds testyuv_cvt.c to the testoverlay2 target, fixing a link error.

slouken committed:
Fixed bug 4034 - Don't include _DllMainCRTStartup() if SDL_STATIC_LIB is defined.

flibitijibibo committed:
Use new XInput mapping for Win10+ (Bugzilla #3960)

Commit:
Fix high-dpi support on macOS and simplify it and iOS variant.

The detault drawableSize for a CAMetalLayer is its bounds x its scale.
So it is sufficient to set the *layer's* scale to the desired value.

icculus committed:
wasapi: let Windows do the resampling for us if possible.

--HG--
extra : histedit_source : d3b8d8c50b1158ecacd48fdd35090ffe24d61c81

icculus committed:
wasapi: fixed typo in an assert message.

--HG--
extra : histedit_source : 50720952e1131985085ddf58c115123988472fb1

icculus committed:
windows: added WIN_IsWindows7OrGreater().

--HG--
extra : histedit_source : cf4acefd794427347dd94a72cb546b4d1da7022c

icculus committed:
vulkan: Possibly fix a compiler warning (-Wstrict-prototypes).

--HG--
extra : rebase_source : 55a8174bb2c9ddf257373ed4034284e9a77b5f51

icculus committed:
android: apparently they fixed this header at some point.

icculus committed:
Backed out changeset bd976b7b2b43

Apparently this still triggers a compiler warning, have to dig further.

BrandonSchaefer committed:
mir: Disable Mir by default as Mir supports Wayland clients

Also remove enabling VK support for Mir

BrandonSchaefer committed:
Backout the vulkan change in cd67e676bca7 breaks Mir builds

slouken committed:
Updated documentation with API changes in SDL 2.0.8

slouken committed:
Fixed MinGW-w64 build

slouken committed:
Added missing file to tvOS build

slouken committed:
Added patch note for Ryan's work

slouken committed:
Fixed bug 4091 - Undefined references to Android audio functions when SDL_AUDIO_DISABLED is on

Manuel Sabogal

If SDL is compiled with the Audio subsystem disabled there are some undefined references to the functions ANDROIDAUDIO_ResumeDevices and ANDROIDAUDIO_PauseDevices in the file src/video/android/SDL_androidevents.c.

slouken committed:
Fixed bug 4092 - CMake support for building everything in the "test" directory

Eric Wasylishen

Patch to support building the tests with cmake.
Disabled by default, use: "cmake .. -DSDL_TEST=YES" to enable the tests.

Tested on macOS 10.13 with the ninja, makefile, and Xcode generators, and Windows 10 with the Visual Studio 2017 generator.

slouken committed:
Moved Android patch note into the Android section

sezero committed:
SDL_test_fuzzer.c: fix strict aliasing warnings by using a union.

suikki committed:
cmake: Set debug library name suffix per target instead of setting it globally. Don't add a suffix on android

slouken committed:
Fixed bug 4097 - Segmentation fault by SDL_CreateThreadWithStackSize

Dongsun Kim

Normal case
1. [thread 1] SDL_CreateThreadWithStackSize calls SDL_SYS_CreateThread.
2. [thread 1] If successful, it calls SDL_SemWait.
3. [thread 2] SDL_RunThread calls SDL_SYS_SetupThread, SDL_ThreadID, SDL_SemPost.
4. [thread 1] SDL_CreateThreadWithStackSize calls SDL_DestroySemaphore, SDL_free.

Crash case (Segmentation fault)
1. [thread 1] SDL_CreateThreadWithStackSize calls SDL_SYS_CreateThread.
2. [thread 1] If successful, it calls SDL_SemWait.
--> Error return due to SIGNAL(SYSTEM or Real Time) at sem_wait(pthread).
3. [thread 1] SDL_CreateThreadWithStackSize calls SDL_DestroySemaphore, SDL_free.
4. [thread 2] SDL_RunThread calls SDL_SYS_SetupThread, SDL_ThreadID, SDL_SemPost.
--> Segmentation fault at strlen or sem_post.

Commit:
Support official Vulkan SDK for macOS.

This tries to load vulkan.framework or libvulkan.1.dylib before MoltenVK.framework
or libMoltenVK.dylib. In the previous version, layers would not work for applications
run-time loading the default library.

slouken committed:
Added patch note about the Vulkan Mac SDK

icculus committed:
assert: Use TerminateProcess() on Windows, vs ExitProcess (thanks, Jack!).

"What I have done is use TerminateProcess rather than ExitProcess.
ExitProcess will cause Microsoft's leak detection to continue, TerminateProcess
won't. It is also technically wrong to use ExitProcess in the case of aborting
the application.

Jack Powell
Twitter @jack9267"

icculus committed:
windows: Message boxes use Task Dialogs if possible (thanks, Jack!).

This lets the message box have an icon. Unless the app has opted-in to using
the v6 common controls, though, this will fall back to the usual SDL message
boxes.

icculus committed:
windows: dos2unix'd messagebox code, and (hopefully) fixed on MingW.

icculus committed:
Back out Task Dialog and TerminateProcess patches for 2.0.8.

These can return to revision control once we ship.

--HG--
extra : histedit_source : a1e341365f4443905266d727b2005654ef1dcb8a%2Ca77cd7cf6988f57299c7e4e4b2f95afbc6ba6c5e

slouken committed:
Added a patch note for OpenGL ES support on Mac

slouken committed:
Temporarily disabled fullscreen switching code on Android, until we can resolve bug 4096 - Enabling fullscreen on Android causes the app to toggle fullscreen mode continuously in a loop

slouken committed:
Added patch note for Android Studio support

slouken committed:
Added tag release-2.0.8 for changeset f1084c419f33

slouken committed:
Use the real controller name for game controllers on iOS and Apple TV

icculus committed:
windows: Restore patches for Task Dialogs and TerminateProcess().

2.0.8 has shipped, these can live in revision control now!

slouken committed:
Fixed setting the layer drawable size
Without this change the drawable had a size of 0 and the metal renderer asserted because the projection matrix wasn't set.

slouken committed:
Fixed bug 4101 - configure needs regenerating after changeset 11894

slouken committed:
Progress fixing bug 4100 - errors and warnings after changeset 11917

Ozkan Sezer 2018-03-02 20:02:37 UTC
http://hg.libsdl.org/SDL/rev/5ce3f8bf8381 resulted in an error and
two warnings when compiled with mingw.

1.  Error from SDL_windowstaskdialog.h:
In file included from src/video/windows/SDL_windowsmessagebox.c:29:0:
src/video/windows/SDL_windowstaskdialog.h:23:54: error: expected ')' before 'HWND'

This is fixed by removing unnecessary annotations:

2.  Warning from SDL_assert.c:
src/SDL_assert.c: In function 'SDL_ExitProcess':
src/SDL_assert.c:138:1: warning: 'noreturn' function does return

Indeed ExitProcess() is prototyped with DECLSPEC_NORETURN, but
TerminateProcess() is not.  This can be rectified by adding an
exit() call in there. Do NOTE, however, that requires building
with a libc:

3.  Warning from SDL_windowsmessagebox.c:
src/video/windows/SDL_windowsmessagebox.c: In function 'WIN_ShowMessageBox':
src/video/windows/SDL_windowsmessagebox.c:513:9: warning: 'nCancelButton' may be used uninitialized in this function

My lazy solution was manually initializing nCancelButton to 0.

Cpasjuste committed:
Merge pull request #12 from spurious/master

merge
